configure.ac

```diff
@@ -1057,7 +1057,10 @@ AS_IF([test x$enable_debug = xyes],
        M0_CPPFLAGS="-DDEBUG $M0_CPPFLAGS"
        M0_KCPPFLAGS="-DDEBUG $M0_KCPPFLAGS"
        M0_CFLAGS="-g -O0 $M0_CFLAGS"
-       M0_KCFLAGS="-g -O0 $M0_KCFLAGS"
+       # For kernel modules compilation, keep the optimization level to -O2 to
+       # fix issue on CentOS 8 and Rocky Linux 8.
+       # Refer https://gcc.gnu.org/bugzilla//show_bug.cgi?id=84187#c3
+       M0_KCFLAGS="-g -O2 $M0_KCFLAGS"
```

Comment:
Agree. Please update the patch!

Reply:
@nikitadanilov, btw, according [to the book](https://www.oreilly.com/library/view/linux-device-drivers/0596000081/ch02s02.html), linux kernel code always must be compiled with at least `-O`:

> A module writer must also specify the -O flag to the compiler, because many functions are declared as inline in the header files. gcc doesn’t expand inline functions unless optimization is enabled, but it can accept both the -g and -O options, allowing you to debug code that uses inline functions.

See also https://stackoverflow.com/a/47123702/3812944.

Reply:
Created https://github.com/Seagate/cortx-motr/pull/1478.

Reply:
Started [PR-1487](https://github.com/Seagate/cortx-motr/pull/1487) to fix the compilation issues for uninitialized variables along with `-Og` option for kernel compilation.